### ISSUE TITLE
Pylance thinks an `object` will never match a dictionary pattern in `match`

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#5963]
## Environment data

-   Language Server version: v2023.9.10
-   OS and version: Manjaro Linux
-   Python version: 3.10.7

## Code Snippet

```python
def f(x: object):
    match x:
        case {"foo": "bar"}:  # <---
            ...
```

## Expected behavior

No error on the marked line

## Actual behavior

> Pattern will never be matched for subject type "object" ([reportUnnecessaryComparison](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUnnecessaryComparison))

This is not true, it is perfectly valid to do `f({"foo": "bar"})`. In my case I'm parsing some JSON, so I think `object` is pretty appropriate.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.328, which I just published. It will also be included in a future release of pylance.
